server/types/date.go

```diff
@@ -120,7 +120,7 @@ func (b DateType) GetSerializationID() SerializationID {
 // IoInput implements the DoltgresType interface.
 func (b DateType) IoInput(ctx *sql.Context, input string) (any, error) {
 	// TODO: need support for calendar era, AD and BC
-	if t, err := time.Parse("2006-01-02", input); err == nil {
+	if t, err := time.Parse("2006-1-2", input); err == nil {
```

Comment:
Are we supposed to support both `2006-01-02` and `2006-1-2`? Or just this new form?

Reply:
Yes, Postgres allows both of these and using `2006-1-2` format works for both